tools/trace.py

```diff
@@ -410,14 +418,17 @@ def generate_program(self, include_self):
 
                 text = heading + """
 {
+        u32 __tgid = bpf_get_current_pid_tgid() >> 32;
+        u32 __pid = bpf_get_current_pid_tgid();
```

Comment:
I don't know if the optimizer can take care of this or not, but I'd probably read into a u64 local and then grab the bottom and top halves from that local.